servers/cu/src/domain/lib/loadMessages.js

```diff
@@ -610,60 +610,98 @@ function loadCronMessagesWith ({ loadTimestamp, findBlocks, loadBlocksMeta, save
           const isColdStart = !ctx.from
 
           /**
-           * Generate and emit a message that represents the process itself.
+           * Generate and emit a message that represents the process itself
+           * if the Process was started before the aop6 Boot Loader change
            *
            * It will be the first message evaluated by the module
            */
+
+          const messages = $messages[Symbol.asyncIterator]()
+
+          /**
+           * { value: any, done: boolean }
+           */
+          let message = await messages.next()
+
+          /**
+           * If this is a cold start
+           *  if (message) then check tags and emit generated if type !== 'Process'
+           *  else convert the Process into a Message and emit it
+           */
           if (isColdStart) {
-            logger('Emitting process message at beginning of evaluation stream for process %s cold start', ctx.id)
-            yield {
-              /**
-               * Ensure the noSave flag is set, so evaluation does not persist
-               * this process message
-               */
-              noSave: true,
-              ordinate: '^',
-              name: `Process Message ${ctx.id}`,
-              message: {
-                Id: ctx.id,
-                Signature: ctx.signature,
-                Data: ctx.data,
-                Owner: ctx.owner,
-                /**
-                 * the target of the process message is itself
-                 */
-                Target: ctx.id,
-                Anchor: ctx.anchor,
+            const { value, done } = message
+            /**
+             * This condition is to handle 2 cases. Before aop6 ECHO Boot loader,
+             * The first Message in a stream will be an actual Message. But after
+             * aop6 the first Message is now the process itself, shaped like a Message
+             *
+             * As a result, old Processes that were started before the boot loader
+             * change, can either have no Messages, or have the first Message with a tag
+             * of type Message, as opposed to Process. In both these cases on a cold start
+             * we need to inject the Process as the first Message in the stream, as was
+             * done prior to the Boot Loader change.
+             */
+            if (done || parseTags(value.Tags).Type !== 'Process') {
```

Comment:
Thinking about it some more, there is an additional case we have to take into account here:

From a cold start, if the first message is a `Cron` message, then we will need to emit the implicitly generated process message. This is to handle the case where an old process was spawned, but emitted a cron message before it's first scheduled message. Should be able to check for `value.message.Cron`

Reply:
also, `value.Tags` needs to actually be `value.message.Tags`

Reply:
See https://github.com/permaweb/ao/blob/b1ecd4d45d174f002388a984acf29c65d6253cd0/servers/cu/src/domain/model.js#L254 for the full shape of `value`